howtos/module/_posts/2016-07-07-Common-mistakes-while-coding-module.md

```diff
@@ -0,0 +1,183 @@
+`Mail::l('Translatable text')`
+
+- Translations inside PDF files:
+`PDF:l('Translatable text')`
+
+Remember to use English as your default language in translations! Indeed, it is always easier to find people to translate from English than other languages.
+
+Note that PrestaShop 1.7 completely [updates the way translation is defined](http://build.prestashop.com/news/new-translation-system-prestashop-17/#calling-symfonys-translator).
+
+
+### 3. Data validation - do not trust your users or yourself!
+
+Recent events show that even the most popular modules or themes can be affected security flow and exploited by hackers. There are a few simple steps which will help you keep your modules secure:
+
+- Always check the input data: use `Tools::getValue('var')` instead of `$_POST['var']`, `$_GET['var']`.
+- [Type hinting](http://php.net/manual/en/language.oop5.typehinting.php) makes sure the value has the correct type: use `(int)`, `(bool)`, `(float)` properly, check the public methods available in the [`classes/Validate.php`](https://github.com/PrestaShop/PrestaShop/blob/develop/classes/Validate.php), such as  [`isUnsignedInt()`](https://github.com/PrestaShop/PrestaShop/blob/develop/classes/Validate.php#L686-L695).
+- The PrestaShop token is your friend: always check the token when doing action for logged in customers. Need an example?
+
+{% gist kpodemski/13a33eba6b43e9b6db7d29e491104ae1 %}
+
+  You can also check the token for guests users: I suggested [these two methods](https://github.com/PrestaShop/PrestaShop/pull/5863/files).
+
+- Secure your files: files outside your module's base file (such as `ajax.php`, `upload.php`, etc.) should have a check at their very beginning in order to make sure they cannot be triggered directly. Here is example:
+
+{% gist kpodemski/5717d7c36d4277a71fa67746bb8efade %}
+
+
+### 4. Setup module compatibility and restrictions
+
+Let's be honest, sometimes it's hard to maintain a module from version to version, and sometimes you'd like to release a module only for specific versions of PrestaShop, even specific countries!
+
+How to set the compatibility of your module? In your `__construct` method:
+
+`$this->ps_versions_compliancy = array('min' => '1.6.1.0', 'max' => _PS_VERSION_);`
+
+This code will not allow to install the module if you have PrestaShop in version lower than 1.6.1.0.
+
+How to restrict your module for specific countries? In your `__construct` method:
+
+`$this->limited_countries = array('fr', 'pl');`
+
+This code will not allow to install the module in countries other than France and Poland. Use [ISO-3166 codes](https://en.wikipedia.org/wiki/ISO_3166-1_alpha-2#Officially_assigned_code_elements)!
+
+
+### 5. Do not trigger code if it is for a specific page
+
+If you are using the `displayHeader` hook to add features to the Product Page (for instance), do not use this code if you're not on page where you have a single product.
+
+You can easily contextualize your code by using simple `if` statements:
```

Comment:
Are the `if` statements simple, or is the solution to simply use `if` statements? The latter would change the sentence into:
> You can easily contextualize your code by simply using `if` statements: